app/controllers/blogposts_controller.rb

```diff
@@ -0,0 +1,75 @@
+class BlogpostsController < ApplicationController
+  before_action :set_blogpost, only: [:edit, :update, :destroy]
+
+  # GET /blogposts
+  # GET /blogposts.json
+  def index
+    @blogposts = Blogpost.all
+  end
+
+  # GET /blogposts/1
+  # GET /blogposts/1.json
+  def show
+    @blogpost = Blogpost.friendly.find(params[:id])
+  end
+
+  # GET /blogposts/new
+  def new
+    @blogpost = Blogpost.new
+  end
+
+  # GET /blogposts/1/edit
+  def edit
+  end
+
+  # POST /blogposts
+  # POST /blogposts.json
+  def create
+    @blogpost = Blogpost.new(blogpost_params)
+
+    respond_to do |format|
+      if @blogpost.save
+        format.html { redirect_to @blogpost, notice: 'Blogpost was successfully created.' }
+        format.json { render :show, status: :created, location: @blogpost }
+      else
+        format.html { render :new }
+        format.json { render json: @blogpost.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /blogposts/1
+  # PATCH/PUT /blogposts/1.json
+  def update
+    respond_to do |format|
+      if @blogpost.update(blogpost_params)
+        format.html { redirect_to @blogpost, notice: 'Blogpost was successfully updated.' }
+        format.json { render :show, status: :ok, location: @blogpost }
+      else
+        format.html { render :edit }
+        format.json { render json: @blogpost.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /blogposts/1
+  # DELETE /blogposts/1.json
+  def destroy
+    @blogpost.destroy
+    respond_to do |format|
+      format.html { redirect_to blogposts_url, notice: 'Blogpost was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_blogpost
+      @blogpost = Blogpost.find(params[:id])
```

Comment:
Maybe it makes sense to change this method to be
`@blogpost = Blogpost.friendly.find(params[:id])` as well (and then put `:show` back in the before action on line 2 and delete your custom code from `def show`). This is more consistent. Otherwise you'll see in the url `blogpost/my-cool-blogpost` but then when you go to edit it it'll be `blogpost/1`. It would be nicer if the url just always stayed sluggified imo. Not a big deal (and I think that FriendlyId can handle this) but could in theory get confusing. 

so line 2 would look like this:
```
before_action :set_blogpost, only: [:show, :edit, :update, :destroy]`
```

the show action would look like this:
```
def show
end
```

and this set blogpost method would look like this:
```
def set_blogpost
   @blogpost = Blogpost.friendly.find(params[:id])
end
```

Reply:
Thanks Tam, consistency wins!